src/main/java/com/commercetools/sync/commons/utils/CustomUpdateActionUtils.java

```diff
@@ -133,6 +136,12 @@
         @Nonnull final Function<T, String> updateIdGetter,
         @Nonnull final BaseSyncOptions syncOptions) {
 
+        // oldResource/newResource types can be different than declared in syncOptions, thus can't
+        // be used in errorCallback; example:
+        // ProductSyncOptions<Product, ProductDraft> but oldResource/newResource can be Asset/AssetDraft
+        boolean resourceMatchWithSyncOptions = ((ParameterizedType) syncOptions.getClass().getGenericSuperclass())
+            .getActualTypeArguments()[0].getClass().isInstance(oldResource);
```

Comment:
What exactly do you mean by _skipping this logic_?
We can always put `null` there, but then we lose the working solution in case the argument types of syncOptions and resource match each other.

Maybe better option is sth like this?
add new parameter to BaseSyncOptions -> `Pair<Class<V>, Class<U>> types` and a method, sth like `public boolean <T extends Custom, S extends CustomDraft> matchingTypes(T oldResource, S newResource)` which will do simple check:
`return types.getKey().isAssignableFrom(newResource.getClass()) && types.getValue()).isAssignableFrom(oldResource.getClass())`

Than we can replace this hackish solution to more future-proof. I'm not saying it is ideal or elegant solution but at least it should somehow solve current problem.